### PR TITLE
Add ability to get version string from libvirt drivers

### PR DESCRIPTION
**Pull request by justinkim**
This PR adds a Versioner interface to `hypervisor` and implements it for `LibvirtDriver` and `RPCDriver`.

cc @benlemasurier

### CODE REVIEW

File: hypervisor/libvirt.go [hunk 1]
```diff
@@ -43,6 +44,21 @@ func (d *LibvirtDriver) DomainNames() ([]string, error) {
...
+		d.prep,
+		d.uri.String(),
+		"version",
+	)
+	if err != nil {
+		return "", err
+	}
+
+	// Strip all newlines down to the version string
+	return strings.TrimSpace(string(out)), nil
```
Comment (mdlayher) [inline hypervisor/libvirt.go]: On my system, this reports:
```
[zsh|matt@nerr-2]:~ 0 % virsh version
Compiled against library: libvirt 1.2.2
Using library: libvirt 1.2.2
Using API: QEMU 1.2.2
Running hypervisor: QEMU 2.0.0
```

Shouldn't you need to process the output to get a version string?

**Comment by mdlayher** [comment]
Add your name under the "Contributors" section in the AUTHORS file!

File: hypervisor/libvirt.go (hunk 1 above)
Comment (justinkim) [inline hypervisor/libvirt.go]: Nice catch. This should be `virsh --version`, instead of `virsh version`.

**Comment by mdlayher** [comment]
:+1:

**Pull request merged**